packages/tenant-process/src/services/tenantService.ts

```diff
@@ -0,0 +1,95 @@
+import { AuthData, eventRepository, initDB } from "pagopa-interop-commons";
+import { Tenant, tenantEventToBinaryData } from "pagopa-interop-models";
+import { v4 as uuidv4 } from "uuid";
+import { TenantProcessConfig } from "../utilities/config.js";
+import {
+  toCreateEventTenantAdded,
+  toCreateEventTenantUpdated,
+} from "../model/domain/toEvent.js";
+import { ApiSelfcareTenantSeed } from "../model/types.js";
+import {
+  assertResourceAllowed,
+  evaluateNewSelfcareId,
+  getTenantKind,
+  getTenantKindLoadingCertifiedAttributes,
+} from "./validators.js";
+import { ReadModelService } from "./readModelService.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function tenantServiceBuilder(
+  config: TenantProcessConfig,
+  readModelService: ReadModelService
+) {
+  const repository = eventRepository(
+    initDB({
+      username: config.eventStoreDbUsername,
+      password: config.eventStoreDbPassword,
+      host: config.eventStoreDbHost,
+      port: config.eventStoreDbPort,
+      database: config.eventStoreDbName,
+      schema: config.eventStoreDbSchema,
+      useSSL: config.eventStoreDbUseSSL,
+    }),
+    tenantEventToBinaryData
+  );
+
+  return {
+    async selfcareUpsertTenant({
+      tenantSeed,
+      authData,
+    }: {
+      tenantSeed: ApiSelfcareTenantSeed;
+      authData: AuthData;
+    }): Promise<string> {
+      const existingTenant = await readModelService.getTenantByExternalId(
+        tenantSeed.externalId
+      );
+      if (existingTenant) {
+        await assertResourceAllowed(existingTenant.data.id, authData);
```

Comment:
the function does not implement the check correctly, it should be changed to

```
export async function assertResourceAllowed(
  resourceId: string,
  authData: AuthData
): Promise<void> {
  const roles = authData.userRoles;
  const organizationId = authData.organizationId;

  if (!roles.includes(userRoles.INTERNAL_ROLE)) {
    return await assertRequesterAllowed(resourceId, organizationId);
  }
}

```

Reply:
> the function does not implement the check correctly, it should be changed to
> 
> ```
> export async function assertResourceAllowed(
>   resourceId: string,
>   authData: AuthData
> ): Promise<void> {
>   const roles = authData.userRoles;
>   const organizationId = authData.organizationId;
> 
>   if (!roles.includes(userRoles.INTERNAL_ROLE)) {
>     return await assertRequesterAllowed(resourceId, organizationId);
>   }
> }
> ```

Sorry, it was my bad on translating it from Scala 🙏

Reply:
Fixed